ios/RCTGoogleAnalyticsBridge/RCTGoogleAnalyticsBridge/Helpers/RNTagEmitterSingleton.m

```diff
@@ -0,0 +1,96 @@
+//
+//  RNTagEmitterSingleton.m
+//  RCTGoogleAnalyticsBridge
+//
+//  Created by Farzad Shafiee on 6/17/18.
+//  Copyright © 2018 Idéhub. All rights reserved.
+//
+
+#import "RNTagEmitterSingleton.h"
```

Comment:
I am not sure this class is needed. I'd rather have `RCTGoogleTagManagerBridge` implement the `TAGFunctionCallTagHandler` and inherit form the `RCTEventEmitter` class.

Reply:
That was my initial implementation, but I couldn't get it working (due to `bridge is not set` error) unless I made the RCTEventEmitter a singleton object. Referencing your [next comment](https://github.com/idehub/react-native-google-analytics-bridge/pull/257#discussion_r197080340) here, I'll give it another shot to see using the `dispatch_async`. hope it works.